File: Backend/SeatifyBackend/Entities/Models/Venue.cs
using Entities.Helpers;
using System.ComponentModel.DataAnnotations;

namespace Entities.Models
{
    public class Venue : IIdEntity
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string Name { get; set; } = string.Empty;

        public string City { get; set; } = string.Empty;

        public string PostalCode { get; set; } = string.Empty;

        public string AddressLine { get; set; } = string.Empty;

        public string OrganizerId { get; set; } = string.Empty;

        public virtual Organizer? Organizer { get; set; }
        public ICollection<Auditorium> Auditoriums { get; set; } = new List<Auditorium>();

        public virtual ICollection<EventOccurrence> EventOccurrences { get; set; } = new List<EventOccurrence>();
    }
}

File: Frontend/SeatifyFrontend/src/app/models/venue.ts
import { Auditorium } from "./auditorium";

export class Venue {
    id: string | undefined = '';
    name: string = '';
    city: string = '';
    postalCode: string = '';
    addressLine: string = '';
    auditoriums: Auditorium[] = [];
    organizerId: string | undefined = '';
}
